src/sagemaker/model_monitor/model_monitoring.py

```diff
@@ -672,15 +672,24 @@ def attach(cls, monitor_schedule_name, sagemaker_session=None):
                 "MonitoringJobDefinition"
             ]["StoppingCondition"].get("MaxRuntimeInSeconds")
 
-        env = schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"]["Environment"]
+        env = schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"].get(
+            "Environment", None
+        )
 
         network_config_dict = schedule_desc["MonitoringScheduleConfig"][
             "MonitoringJobDefinition"
         ].get("NetworkConfig")
 
-        vpc_config = schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"][
-            "NetworkConfig"
-        ].get("VpcConfig")
+        vpc_config = None
+        if (
+            schedule_desc["MonitoringScheduleConfig"]["MonitoringJobDefinition"].get(
```

Comment:
should this be just ``if network_config_dict is not None``

Reply:
I would also put all the network_config parsing stuff in the same if block.

Reply:
That makes sense! I also refactored the `attach` method to make it more concise